i18n/en.json

```diff
@@ -4990,6 +4994,14 @@
     "id": "model.incoming_hook.id.app_error",
     "translation": "Invalid Id"
   },
+  {
+    "id": "model.incoming_hook.username.app_error",
+    "translation": "Invalid username"
```

Comment:
This "duplicates" `Invalid username.` above, but this seemed to be the pattern. (I'd much prefer to see us embedding the English messages in the source code and have a tool to extract strings for translation, as it's cumbersome to manage this file.)

Note too that I've omitted the period in these two messages, unlike above, to match the format of the surrounding error messages.

Reply:
I get what you mean and we had such tool but in the end it ended up being way easier to handle it this way and changing it at this point In time back to how it was is going to break all the translations, so I don’t really see the need to do that.

Reply:
Absolutely, I can appreciate that I don't have the historical context for these changes. :)

One concern with the current approach, though, is a typo in the identifier, such as https://github.com/mattermost/mattermost-server/blob/3a4316a235616d6251308cf3a7d0b8fe2176d6f1/i18n/en.json#L1424. `api.incoming_webhook.disabled.app_errror` doesn't appear to be used anywhere, and it's correctly typed counterpart `api.incoming_webhook.disabled.app_error`, referenced in the go code, doesn't appear to actually have a translation at the moment. I've submitted a separate PR to address this: https://github.com/mattermost/mattermost-server/pull/8009

One way to help with this and stick with the current approach of using identifiers might be to define them as constants. Although that just moves the potential for a typo downstream, it might be easier to prevent being in just one place. I'd be happy to chat about other ways we could improve things here too :)